PracticeRobot1/robot1/src/main/java/frc/robot/Robot.java

```diff
@@ -0,0 +1,203 @@
+ * creating this project, you must also update the build.gradle file in the
+ * project.
+ */
+public class Robot extends TimedRobot {
+
+  private final DifferentialDrive m_robotDrive = new DifferentialDrive(new PWMVictorSPX(0), new PWMVictorSPX(1));
+  private final Joystick m_stick = new Joystick(0);
+  private final Timer m_timer = new Timer();
+
+  private static final String kDefaultAuto = "Default";
+  private static final String kCustomAuto = "My Auto";
+  private String m_autoSelected;
+  private final SendableChooser<String> m_chooser = new SendableChooser<>();
+
+  // Encoders
+  private final Encoder m_leftEncoder = new Encoder(0, 1, 2);
+  private final Encoder m_rightEncoder = new Encoder(3, 4, 5);
+
+  // Gyro
+  private final AnalogGyro m_gyro = new AnalogGyro(1);
+
+  // Simulation Stuff
+  private final DifferentialDriveOdometry m_odometry = new DifferentialDriveOdometry(
+      Rotation2d.fromDegrees(getHeading()), new Pose2d(4, 5, new Rotation2d()));;
+  private EncoderSim m_leftEncoderSim;
+  private EncoderSim m_rightEncoderSim;
+  private Field2d m_fieldSim;
+  private AnalogGyroSim m_gyroSim;
+  public DifferentialDrivetrainSim m_driveTrainSim;
+
+  /**
+   * This function is run when the robot is first started up and should be used
+   * for any initialization code.
+   */
+  @Override
+  public void robotInit() {
+    m_chooser.setDefaultOption("Default Auto", kDefaultAuto);
+    m_chooser.addOption("My Auto", kCustomAuto);
+    SmartDashboard.putData("Auto choices", m_chooser);
+
+    // Set Encoder pulses
+    m_leftEncoder.setDistancePerPulse((0.1524 * Math.PI) / (double) 1024);
+    m_rightEncoder.setDistancePerPulse((0.1524 * Math.PI) / (double) 1024);
+    m_leftEncoder.reset();
+    m_rightEncoder.reset();
+
+    if (!RobotBase.isReal()) {
+      // Set up robot simulation
+      m_driveTrainSim = DifferentialDrivetrainSim.createKitbotSim(KitbotMotor.kDualCIMPerSide, KitbotGearing.k10p71,
+          KitbotWheelSize.SixInch, null);
+      m_fieldSim = new Field2d();
+      SmartDashboard.putData("Field", m_fieldSim);
+
+      // Connect the simulators with their counterparts
+      m_leftEncoderSim = new EncoderSim(m_leftEncoder);
+      m_rightEncoderSim = new EncoderSim(m_rightEncoder);
+      m_gyroSim = new AnalogGyroSim(m_gyro);
+    }
+  }
+
+  /**
+   * This function is called every robot packet, no matter the mode. Use this for
+   * items like diagnostics that you want ran during disabled, autonomous,
+   * teleoperated and test.
+   *
+   * <p>
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and SmartDashboard integrated updating.
+   */
+  @Override
+  public void robotPeriodic() {
+  }
+
+  @Override
+  public void simulationPeriodic() {
+
+    // connect the motors to update the drivetrain
+    m_driveTrainSim.setInputs(m_leftMotors.get() * RobotController.getBatteryVoltage(),
+        m_rightMotors.get() * RobotController.getBatteryVoltage());
```

Comment:
Make sure to change m_leftMotors and m_rightMotors to the speed controller names you are using